lib/ood_core/job/adapters/ccq.rb

```diff
@@ -0,0 +1,300 @@
+          script_file = make_script_file(script.content)
+          args = []
+
+          # cluster configuration args
+          args.concat ["-s", scheduler] unless scheduler.nil?
+          args.concat [image_arg, image] unless image.nil?
+
+          args.concat ["-o", script.output_path.to_s] unless script.output_path.nil?
+          args.concat ["-e", script.error_path.to_s] unless script.error_path.nil?
+          args.concat ["-tl", seconds_to_duration(script.wall_time)] unless script.wall_time.nil?
+          args.concat ["-js", script_file.path.to_s]
+
+          args.concat script.native if script.native
+
+          output = call("ccqsub", args: args)
+          parse_job_id_from_ccqsub(output)
+        ensure
+          script_file.close
+        end
+
+        # Retrieve info for all jobs from the resource manager
+        # @return [Array<Info>] information describing submitted jobs
+        def info_all(attrs: nil)
+          args = []
+          args.concat ["-s", scheduler] unless scheduler.nil?
+
+          stat_output = call("ccqstat", args: args)
+          info_from_ccqstat(stat_output)
+        end
+
+        # Retrieve info for all jobs for a given owner or owners from the
+        # resource manager
+        # @param owner [#to_s, Array<#to_s>] the owner(s) of the jobs
+        # @return [Array<Info>] information describing submitted jobs
+        def info_where_owner(owner, attrs: nil)
+          owner = Array.wrap(owner).map(&:to_s)
+
+          # must at least have job_owner to filter by job_owner
+          attrs = Array.wrap(attrs) | [:job_owner] unless attrs.nil?
+
+          info_all(attrs: attrs).select { |info| owner.include? info.job_owner }
+        end
+
+        # Retrieve job info from the resource manager
+        # @param id [#to_s] the id of the job
+        # @return [Info] information describing submitted job
+        def info_all_each(attrs: nil)
+          return to_enum(:info_all_each, attrs: attrs) unless block_given?
+
+          info_all(attrs: attrs).each do |job|
+            yield job
+          end
+        end
+
+        # Retrieve job info from the resource manager
+        # @param id [#to_s] the id of the job
+        # @return [Info] information describing submitted job
+        def info(id)
+          args = []
+          args.concat ["-s", scheduler] unless scheduler.nil?
+          args.concat ["-ji", id]
+
+          stat_output = call("ccqstat", args: args)
+
+          # WARNING: code path differs here than info_all because the output
+          # from ccqstat -ji $JOBID is much more data than just the 4
+          # columns that ccqstat gives.
+          info_from_ccqstat_extended(stat_output)
+        end
+
+        # Retrieve job status from resource manager
+        # @param id [#to_s] the id of the job
+        # @return [Status] status of job
+        # @see Adapter#status
+        def status(id)
+          info(id).status
```

Comment:
If there is a `ccqstat` that returns less information than the info that might be preferred but that is a future optimization

Reply:
There is, but it's unclear whether it'd actually be faster or not because while it is less data about any _one_ job, it is data for _all_ jobs (also `completed` state sticks around until deleted by a user).